agent/agents/postgres/pgstatmonitor/pgstatmonitor.go

```diff
@@ -241,7 +241,7 @@ func (m *PGStatMonitorQAN) Run(ctx context.Context) {
 	if current, _, err := m.monitorCache.getStatMonitorExtended(ctx, m.q, normalizedQuery); err == nil {
 		m.monitorCache.refresh(current)
 		m.l.Debugf("Got %d initial stat monitor.", len(current))
-		running = true
+		running = true //nolint:ineffassign //TODO: @Nurlan please check this warning, it seems that we may have a bug here.
```

Comment:
@BupycHuk please take a look

Reply:
looks like it's in a right place, but the next one which overwrites it is incorrect.

Reply:
I think the line 255 should be
```
running = running && m.checkDefaultWaitTime(waitTime)
```

Reply:
I applied suggested fix, but whole this chunk looks strange, at first we may send `running` state, and just after that switch it back to `waiting`. It looks like we can at first check all conditions, and then send status.

Reply:
agree

Reply:
Good idea. Thanks.

@BupycHuk is it possible to drop older versions and have compatibility only with PGSM 1.0 and higher? In next PMM versions. Code will be shorter and cleaner then and because of problems in older  PGSM versions (like wrong metrics etc) noone should use it anyway.